src/session_impl.cpp

```diff
@@ -5162,11 +5175,14 @@ namespace aux {
 		// proxy, and it's the same one as we're using for the tracker
 		// just tell the tracker the socks5 port we're listening on
 		if (m_socks_listen_socket && m_socks_listen_socket->is_open())
-			return m_socks_listen_socket->local_endpoint().port();
+		{
+			error_code ec;
+			return m_socks_listen_socket->local_endpoint(ec).port();
```

Comment:
is ignoring the error here the right thing to do? would you expect this to legitimately fail? without it being serious?